### PR TITLE
Add ability to set the start and endtime of the graph data

### PR DESCRIPTION
**Pull request by hvangeffen**
Date input currently shows 13/06/2024 as it is formatted according to:
```ts
new Intl.DateTimeFormat(LOCALE_SET_FOR_VUETIFY, { 
    year: 'numeric', month: '2-digit', day: '2-digit'
}).format(date)
```

### CODE REVIEW

File: src/lib/TimeControl/interval.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { TimeSettingsViewPeriodPreset } from '@deltares/fews-pi-requests'
```
Comment (ceesvoesenek) [inline src/lib/TimeControl/interval.ts:1]: This can also be `import type`.

File: src/services/useDisplayConfig/index.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { DisplayConfig } from '../../lib/display/DisplayConfig.js'
 import { timeSeriesDisplayToChartConfig } from '../../lib/charts/timeSeriesDisplayToChartConfig.js'
 import { createTransformRequestFn } from '@/lib/requests/transformRequest.js'
 import { MD5 } from 'crypto-js'
+import { convertFewsPiDateTimeToJsDate } from '@/lib/date/index.js'
```
Comment (ceesvoesenek) [inline src/services/useDisplayConfig/index.ts:13]: Remove the `index.js` (probably auto-import?)

File: src/services/useDisplayConfig/index.ts [hunk 3]
```diff
@@ -41,16 +42,34 @@ function isTimeSeriesGridActionsFilter(
...
+    if (period) {
+      const periodStart = convertFewsPiDateTimeToJsDate(
+        period.startDate,
+        timeZoneOffsetString,
+      )
+      const periodEnd = convertFewsPiDateTimeToJsDate(
+        period.endDate,
+        timeZoneOffsetString,
+      )
+      configPeriod = [startTime ?? periodStart, endTime ?? periodEnd]
```
Comment (ceesvoesenek) [inline src/services/useDisplayConfig/index.ts:67]: We don't need to convert the FEWS PI date/times if we already know we're not going to use them?

File: src/services/useDisplayConfig/index.ts [hunk 4]
```diff
@@ -87,14 +108,29 @@ export function useDisplayConfig(
   const displayConfig = ref<DisplayConfig>()
   const displays = ref<DisplayConfig[]>()
 
+  const response = ref<ActionsResponse>()
+
   watchEffect(async () => {
```
Comment (ceesvoesenek) [inline src/services/useDisplayConfig/index.ts:113]: Maybe add a brief comment on why the fetch is separated from the interpretation of the response?

File: src/services/useDisplayConfig/index.ts [hunk 5]
```diff
@@ -117,24 +153,25 @@ export function useDisplayConfig(
...
-    let nodeId = undefined
     if (isFilterActionsFilter(_filter)) {
       if (!_filter.filterId) return
-      response = await piProvider.getFilterActions(_filter)
+      response.value = await piProvider.getFilterActions(_filter)
     } else if (isTimeSeriesGridActionsFilter(_filter)) {
-      response = await piProvider.getTimeSeriesGridActions(_filter)
-      response.results.forEach((result) => {
+      response.value = await piProvider.getTimeSeriesGridActions(_filter)
+      response.value.results.forEach((result) => {
```
Comment (ceesvoesenek) [inline src/services/useDisplayConfig/index.ts:174]: Not your code, and I haven't looked in too much detail, but could this just be `flatMap`?

**Review by ceesvoesenek — COMMENTED** [review]
Good work, seems to have been an annoying problem! Thanks for fixing my mess too 😉 

Some comments, but nothing dramatic

**Review by ekkelenkamp — APPROVED** [review]
LGTM

**Pull request merged**